src/main/java/org/redhat/sbomer/rest/v1alpha1/SBOMResource.java

```diff
@@ -116,50 +99,77 @@ public Page<Sbom> list(@Valid @BeanParam PaginationParameters paginationParams)
     @Parameter(name = "buildId", description = "PNC build identifier", example = "ARYT3LBXDVYAC")
     @APIResponses({
             @APIResponse(
                     responseCode = "200",
-                    description = "The BaseSBOM structure for a specific PNC buildId.",
+                    description = "The SBOMs related to a specific PNC buildId.",
                     content = @Content(mediaType = MediaType.APPLICATION_JSON)),
             @APIResponse(
                     responseCode = "404",
-                    description = "The BaseSBOM for the particular buildID couldn't be found in the system.",
+                    description = "No SBOMs could be found for the specified PNC build.",
                     content = @Content(mediaType = MediaType.APPLICATION_JSON)) })
-    public Sbom get(@PathParam("buildId") String buildId) {
-        return sbomService.getBaseSbom(buildId);
+    public Page<Sbom> listAllWithPncBuildId(
+            @PathParam("buildId") String buildId,
+            @Valid @BeanParam PaginationParameters paginationParams) {
+        return sbomService
+                .listAllSbomsWithBuildId(buildId, paginationParams.getPageIndex(), paginationParams.getPageSize());
+    }
+
+    @POST
+    @Operation(
+            summary = "Generate a base SBOM based on the PNC build using Domino.",
+            description = "SBOM base generation for a particular PNC build Id offloaded to the service, using Domino.")
+    @Parameter(name = "buildId", description = "PNC build identifier", example = "ARYT3LBXDVYAC")
+    @Path("/generate/domino/{buildId}")
+    @APIResponses({ @APIResponse(
+            responseCode = "201",
+            description = "Schedules generation of a SBOM for a particular PNC buildId. This is an asynchronous call. It does execute the generation behind the scenes.",
+            content = @Content(mediaType = MediaType.APPLICATION_JSON)) })
+    public Response generateFromBuildWithDomino(@PathParam("buildId") String id) throws Exception {
+
+        return sbomService.generateSbomFromPncBuild(id, Generators.DOMINO);
     }
 
     @POST
     @Operation(
-            summary = "Enrich SBOM based on the PNC build",
-            description = "SBOM enrichment for a particular PNC build Id. The only mode currently available is `ENRICHED_v1_0`")
+            summary = "Generate a base SBOM based on the PNC build using CycloneDX plugin.",
+            description = "SBOM base generation for a particular PNC build Id offloaded to the service, using CycloneDX plugin.")
     @Parameter(name = "buildId", description = "PNC build identifier", example = "ARYT3LBXDVYAC")
-    @Parameter(
-            name = "mode",
-            description = "Enrichment generation mode. Available modes: `ENRICHED_v1_0`",
-            example = "ENRICHED_v1_0")
-    @Path("/enrich/{buildId}")
+    @Path("/generate/cyclonedx/{buildId}")
```

Comment:
I wonder if we are interested in multiple endpoints like this or maybe it would be better to expose one: `/generate`. It could take an optional `@PathParam`. By default it would be `CYCLONEDX`, but if needed, it could be overriden. This way we would make it more flexible to add/remove generators. The other benefit is that we would actually point users which one is preferred (default). Not sure if they really care about what is the implementation.